pkg/kvstore/etcd.go

```diff
@@ -101,12 +108,60 @@ func NewEtcdClient(config *client.Config, cfgPath string) (KVClient, error) {
 				ec.session = newSession
 				ec.sessionMU.Unlock()
 				log.Debugf("Renewing etcd session")
+				if err := ec.CheckMinVersion(10 * time.Second); err != nil {
+					log.Fatalf("%s", err)
```

Comment:
>We do this in an infinite loop because the etcd version could change while Cilium is running, so we need to consistently keep checking?

Yes. The user can add a new etcd node running `3.0.0` in the etcd cluster. That's a no go :)